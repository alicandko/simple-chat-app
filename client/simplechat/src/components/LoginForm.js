import React, { Component } from 'react';

export class LoginForm extends Component {
  state = {
    username: null,
  };

  handleSubmit = (event) => {
    event.preventDefault();
  };

  handleChange = (event) => {
    this.setState({
      username: event.target.value,
    });
  };

  render() {
    const { username } = this.props;
    return (
      <form className="login-form form" onSubmit={this.handleSubmit}>
        <label className="login-form__label" htmlFor="username">
          Username
        </label>
        <input
          type="text"
          className="login-form__username-input"
          id="username"
          onChange={this.handleChange}
          value={username}
          placeholder="Enter username"
        />
        <input
          type="submit"
          className="login-form__button"
          disabled={!this.state.username}
          value="Connect"
        />
      </form>
    );
  }
}
