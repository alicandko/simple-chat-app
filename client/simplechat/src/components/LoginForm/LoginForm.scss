.login-form {
  margin: 10px;
  width: 100%;
  display: flex;
  flex-direction: column;
}
