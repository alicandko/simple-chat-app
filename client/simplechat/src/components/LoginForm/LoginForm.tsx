import React, { Component } from 'react';

import './LoginForm.scss';

interface Props {
  connectRequested: (username: string) => void;
}

interface State {
  username: string;
}

export class LoginForm extends Component<Props, State>  {
  public state: State = {
    username: '',
  };

  public handleSubmit = (event) => {
    event.preventDefault();
    this.props.connectRequested(this.state.username);
  };

  public handleChange = (event) => {
    this.setState({
      username: event.target.value,
    });
  };

  public render() {
    const { username } = this.state;
    return (
      <form className="login-form form" onSubmit={this.handleSubmit}>
        <label className="login-form__label" htmlFor="username">
          Username
        </label>
        <input
          type="text"
          className="login-form__username-input text-input"
          id="username"
          onChange={this.handleChange}
          value={username}
          placeholder="Enter username"
        />
        <input
          type="submit"
          className="login-form__button button"
          disabled={!this.state.username}
          value="Connect"
        />
      </form>
    );
  }
}
