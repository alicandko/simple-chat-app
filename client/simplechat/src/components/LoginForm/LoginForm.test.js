import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { LoginForm } from './LoginForm';

const mockFunction = jest.fn();

describe('CheckboxInput', () => {
  it('should render correctly', () => {
    const component = renderer
      .create(<LoginForm username="" connectRequested={mockFunction} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should have the correct passed props', () => {
    const wrapper = mount(<LoginForm connectRequested={mockFunction} />);

    expect(wrapper.prop('connectRequested')).toEqual(mockFunction);
  });

  describe('when button is clicked', () => {
    it('should execute the connectRequested function', () => {
      const wrapper = mount(<LoginForm connectRequested={mockFunction} />);

      wrapper.find('.login-form__button').simulate('click');

      expect(mockFunction.mock.calls.length).toEqual(0);
    });
  });

  describe('when enter is pressed', () => {
    it('should execute the connectRequested function', () => {
      const wrapper = mount(<LoginForm connectRequested={mockFunction} />);

      wrapper.find('.login-form__button').simulate('keypress', { key: 'Enter' });

      expect(mockFunction.mock.calls.length).toEqual(0);
    });
  });
});
