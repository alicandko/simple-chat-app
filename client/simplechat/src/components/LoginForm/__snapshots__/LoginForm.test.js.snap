// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxInput should render correctly 1`] = `
<form
  className="login-form form"
  onSubmit={[Function]}
>
  <label
    className="login-form__label"
    htmlFor="username"
  >
    Username
  </label>
  <input
    className="login-form__username-input"
    id="username"
    onChange={[Function]}
    placeholder="Enter username"
    type="text"
    value=""
  />
  <input
    className="login-form__button"
    disabled={true}
    type="submit"
    value="Connect"
  />
</form>
`;
